helm/prometheus-agent/charts/prometheus-agent/files/watchdog.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+
+### Config
+
+metrics_url="localhost:9090/metrics"
+overload_tolerance="${WATCHDOG_OVERLOAD_TOLERANCE:-2}" # allow up to x times max shards
+
+
+exit_error() {
+    echo "ERROR: $*"
+    exit 1
+}
+
+### Default liveness probe:
+#    livenessProbe:
+#      failureThreshold: 6
+#      httpGet:
+#        path: /-/healthy
+#        port: http-web
+#        scheme: HTTP
+#      periodSeconds: 5
+#      successThreshold: 1
+#      timeoutSeconds: 3
+default_liveness() {
+  wget --timeout 3 --tries 3 -q -O /dev/null http://localhost:9090/-/ready
+  return $?
+}
+
+
+main() {
+
+    default_liveness \
+        || exit_error "liveness probe failed"
+
+    shards_desire="$(wget -O- --timeout 2 --tries 3 -q "$metrics_url" \
+                     | sed -n 's/^prometheus_remote_storage_shards_desire.* \([0-9]*\)\.[0-9]*/\1/p')"
+    [ -z "$shards_desire" ] \
+        && exit_error "could not determine desired shards"
```

Comment:
I assume if `/metrics` does not return valid data, there's something wrong with our prometheus-agent.

Reply:
Agreed that we prefer to have the pod being killed repeatedly when this script fails (maybe the metrics name changed, maybe something changes in the agent), rather than have it fails silently.
So someone will actually go and check this instead of loosing the watchdog functionality without knowing.